subsys/fs/nvs/nvs.c

```diff
@@ -781,10 +781,20 @@ static int nvs_startup(struct nvs_fs *fs)
 			}
 		}
 	}
-	/* all sectors are closed, this is not a nvs fs */
+	/* all sectors are closed, this is not a nvs fs or irreparably corrupted */
 	if (closed_sectors == fs->sector_count) {
+#ifdef CONFIG_NVS_INIT_BAD_REGION
+		LOG_WRN("All sectors closed, erasing all sectors...");
+		addr = 0U;
```

Comment:
Was this code tested? If you enter this `if` and you erase the sectors, then you also enter the `if (i == fs->sector_count)` condition below, where there is `rc = nvs_flash_cmp_const(fs, addr - ate_size, erase_value, sizeof(struct nvs_ate));` that assumes that `addr` points to the last ATE within the last sector. Am I correct?

Please consider extending NVS unit tests with a scenario that you initialize NVS using a garbage data in all sectors.

Reply:
Just did a full re-test with the latest version I pushed. Set the ATEs to 0xdeadbeef and verified the log is printed and the NVS settings are written correctly afterwards.

Can you point me to the unit tests for NVS? I don't see them in this directory.

Reply:
Thank you. The tests are in `tests/subsys/fs/nvs` directory. Though feel free to merge this PR and add the tests later :)